### PR TITLE
fix: deploy headless.js file for tier1 environment

### PR DESCRIPTION
**Pull request by bc-marco**
Jira: [B2B-1009](https://bigcommercecloud.atlassian.net/browse/B2B-1009)

## What/Why?

Fix for deployment process to tier1

## Rollout/Rollback

Undo this PR

## Testing
image

[B2B-1009]: https://bigcommercecloud.atlassian.net/browse/B2B-1009?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by bc-micah** [comment]
@LeoChowChina - please review this PR. we need to fix this before we run tier 1 deployment (or merge other PRs).

File: apps/storefront/vite.config.ts [hunk 1]
```diff
@@ -10,13 +10,20 @@ interface AssetsAbsolutePathProps {
...
   staging: 'https://cdn.bundleb2b.net/b2b/staging/storefront/assets/',
   production: 'https://cdn.bundleb2b.net/b2b/production/storefront/assets/',
 };
 
 export default defineConfig(({ mode }) => {
-  const env = loadEnv(mode, process.cwd());
+  let modeVariable = mode;
+  if (mode === TIER1_ENV) {
+    modeVariable = PRODUCTION_ENV;
+  }
```
Comment (BrianJiang2021) [inline apps/storefront/vite.config.ts:25]: const modeVariable = (mode === TIER1_ENV) ? PRODUCTION_ENV : mode;
Is it more concise and readable?

**Review by LeoChowChina — APPROVED** [review]
looks good to me

**Comment by BrianJiang2021** [comment]
The app should have a corresponding fix: https://github.com/bigcommerce/b2b-merchant-dashboard/pull/2698

File: apps/storefront/vite.config.ts [hunk 2]
```diff
@@ -100,8 +107,12 @@ export default defineConfig(({ mode }) => {
         },
         output: {
           entryFileNames(info) {
+            let baseNameFile = '[name].js';
+            if (mode === TIER1_ENV) {
+              baseNameFile = '[name]-tier1.js';
+            }
             const { name } = info;
-            return name.includes('headless') ? '[name].js' : '[name].[hash].js';
+            return name.includes('headless') ? baseNameFile : '[name].[hash].js';
```
[the remark this answers is not in the document]
Comment (bc-micah) [reply apps/storefront/vite.config.ts:115]:
```
entryFileNames({name}) {
  if (name.includes('headless')) {
     return mode === TIER1_ENV ?
        '[name]-tier1.js' :
        '[name].js';
  }
  
  return '[name].[hash].js';
}
```

**Review by bc-micah — APPROVED** [review]

**Pull request merged**